IoT app updates

bbartling opened the pull request:
I added this to the README and also included the Yolo v7 tiny model into the repo....it seems small enough that Github wont discard a push.

```
## 7. Run IoT web app to count people with yolov7-tiny model
This is a feature to run `webcam.py` to render computer vision results in the browser. 
The idea for web browser access is for remote deployments purposes and for convenience purposes use a browser to monitor results if the remote computer can be accessed via VPN or port forwarding. 
The web app also contains a rest endpoint for people and video frame rate which can be logged on an external source requesting locally data via a GET requests to the web app.

* To run IoT app `cd` into `python` and use an additional `arg` to use flask
* `$ python webcam.py -i 0 -m ./models/yolov7-tiny.onnx --use-flask`
* Open browser dial into `localhost:5000` or the IP of the computer on port 5000 after setting up appropriate firewall rules
* People count rest API GET request: `http://localhost:5000/people-count/` to log data from external IoT platform
* FPS rest API GET request: `http://localhost:5000/fps/` to log data from external IoT platform
* Future testing to include MQTT if desired
```

Curious on your thoughts : )

OpenVINO-dev-contest reviewed (commented):
Hi @bbartling really appreciated for your contribution. Could you also help to check is there any extra dependency may be needed to run this demo, and we can add them into **python/requirements.txt**

bbartling commented:
Ah yes I forgot to add `flask`. The only other thing is there is a templates directory for the html file...and also a webapp_utils.py`.

OpenVINO-dev-contest commented:
Hi @bbartling thanks again for your contribution. 
The detection results streaming works well on my system, and data from `http://localhost:5000/people-count/` and `http://localhost:5000/fps/` shows correctly, however the counting data in `http://localhost:5000` is missing. Could you help to check ?
<img width="956" alt="yolov7_flask" src="https://user-images.githubusercontent.com/91237924/208337023-07c399ca-10d5-40c3-b1dd-a8eb7ba0788c.png">

bbartling commented:
So it looks like you have the correct html file located in the templates directory by your screenshot I can tell by the chart on the right hand side was my last commit [ad900b0](https://github.com/OpenVINO-dev-contest/YOLOv7_OpenVINO_cpp-python/pull/8/commits/ad900b0a4a7f9a32799164116950bf70e4d9e6bf)

Is there any chance you could try just clearing your browser cache and try again? On `localhost:5000` if its the same computer running the app.

![image](https://user-images.githubusercontent.com/29737117/208442680-5f489fea-bd78-4a38-ae19-ab6e6923fecb.png)


1. After clearing cache its not working the JavaScript inside the browser has an issue...in your browser dev tools can you screenshot the `console`? These are the JavaScript functions that hit the web app endpoints to render widgets/chart:
![image](https://user-images.githubusercontent.com/29737117/208443759-de2a0629-893e-4b93-8abd-950bc56a4917.png)

There maybe some error popping up in the console logs

OpenVINO-dev-contest commented:
Sorry, Im not much familiar with js and frontend development,  and it seems caused by `undefined $` ?
<img width="913" alt="image" src="https://user-images.githubusercontent.com/91237924/208461880-b3e5aa25-9ca2-4c15-ac54-00b4ef53d8cd.png">

<img width="411" alt="image" src="https://user-images.githubusercontent.com/91237924/208462446-6d814a73-e320-4754-bcbf-7b6f071859e0.png">

bbartling commented:
Did you clear your browser cache? Also curious if you could try another browser like edge or firefox

OpenVINO-dev-contest commented:
Yes, and I try both on Chrome with windows and Firefox with Ubuntu.  Any way I will ask my friends to try it later.

bbartling commented:
The `$` not being defined is the JavaScript resources **_aren't being pulled into the browser from the internet_**. These top 13 lines of the html file defines how your browser reaches out to grab the JavaScript library from the web:
```
<!doctype html>
<html lang="en">

<head>
    <title>YoloV7 IoT Live Stream</title>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet">
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js"></script>
</head>
<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.4.2/jquery.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/chart.js"></script>
<script src="https://cdn.jsdelivr.net/npm/luxon@3.0.4/build/global/luxon.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/chartjs-adapter-luxon@1.2.0/dist/chartjs-adapter-luxon.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/chartjs-plugin-streaming@2.0.0/dist/chartjs-plugin-streaming.min.js"></script>
```

Can you verify these first 13 lines of the JavaScript in the html file if the clearing the cache in the browser doesn't fix the issue? 

My fork:
```
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
```

OpenVINO-dev-contest commented:
I think there is the connection issue on accessing `http://ajax.googleapis.com/ajax/libs/jquery/1.4.2/jquery.min.js` in PRC

OpenVINO-dev-contest commented:
Could you help to solve the conflict, and I will merge the PR, thanks

bbartling commented:
Do these sources load?

- http://ajax.microsoft.com/ajax/jquery/jquery-1.4.2.min.js
- https://cdnjs.cloudflare.com/ajax/libs/jquery/1.4.2/jquery.min.js

Short term solution ill fix the script to load a local version of the 1.4.2 JavaScript file.

In the long term future I am learning VUE JavaScript but it make take me a bit to figure out how to overhaul the older outdated jQuery library.

bbartling commented:
I may just overhaul this to VUE JavaScript....give me a few weeks : )

OpenVINO-dev-contest commented:
thanks, its OK i think if you wish to merge it now. And I got COVID this week, so pls forgive that the reply from me will be late.

bbartling commented:
Sounds good rest up!

bbartling commented:
Hi @OpenVINO-dev-contest is there anyway you could give me a tip on how to merge my fork into yours? I actually never have this before on Git

OpenVINO-dev-contest commented:
> Hi @OpenVINO-dev-contest is there anyway you could give me a tip on how to merge my fork into yours? I actually never have this before on Git

Hi You have to solve the conflicting files and push again. Then I can merge it.
<img width="352" alt="image" src="https://user-images.githubusercontent.com/91237924/209678702-9c8f74a1-22e4-4259-aad4-d4a18d7f7a4a.png">

bbartling commented:
Hi @OpenVINO-dev-contest sorry for making this into a hassle...but for the 3 conflicts, the README I just moved your bullet point number 7 to number 9 in my README so I could include a few more IoT setup info's as shown below on the `Further optimization` link.

```
=======
Tips: you can switch the device name to **"GPU"** to improve the performance.

## 9. Further optimization
Try this notebook ([yolov7-optimization](https://github.com/openvinotoolkit/openvino_notebooks/tree/develop/notebooks/226-yolov7-optimization)) and quantize your YOLOv7 model to INT8.


=======
Tips: you can switch the device name to **"GPU"** to improve the performance.

## 7. Further optimization
Try this notebook ([yolov7-optimization](https://github.com/openvinotoolkit/openvino_notebooks/tree/develop/notebooks/226-yolov7-optimization)) and quantize your YOLOv7 model to INT8.
>>>>>>> main
```

And also the conflicts for:
```
python/webcam.py
python/yolov7.py
```

These are for passing additional `args` from arg parser to yolo7.py for flask and also an arg for confidence.

Thanks for any tips....first time for me using github to merge a fork...my background is all automation PLC programming so this world is a bit new to me : ) Also hope the COVID recovery is going well!! I think I also went through that about a month ago but it could have been the RSV virus..anyways let me know if I should something different in this process!!

OpenVINO-dev-contest commented:
hi @bbartling I think its the issues cased by version conflict. You can try to rebase your repository with ```git rebase``` and solve the conflict, or follow my tips to recreate the PR:
Backup your changes, and then sync-up the whole repository. here is some step for forcing sync-up the local repository.
```
git fetch --all
git reset --hard origin/master
git pull
```
After that, you can create the changes again and push the PR.

bbartling commented:
On my end I get an error on the `git reset --hard`...any ideas to try? look like this:

```
 $ git fetch --all
Fetching origin
```

Next one: `$ git reset --hard origin/master`
returns this:
```
fatal: ambiguous argument 'origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

OpenVINO-dev-contest commented:
> On my end I get an error on the `git reset --hard`...any ideas to try? look like this:
> 
> ```
>  $ git fetch --all
> Fetching origin
> ```
> 
> Next one: `$ git reset --hard origin/master` returns this:
> 
> ```
> fatal: ambiguous argument 'origin/master': unknown revision or path not in the working tree.
> Use '--' to separate paths from revisions, like this:
> 'git <command> [<revision>...] -- [<file>...]'
> ```

Try ```git reset --hard origin/main``` ?

bbartling commented:
@OpenVINO-dev-contest 

So I was able to do this now....thanks for the newbie tip!

I did this on my end:
git fetch --all
git reset --hard origin/`main`
git pull

and it all work : )

bbartling commented:
Hi @OpenVINO-dev-contest let me know if I can do anything on my end to merge in the IoT stuff : )

OpenVINO-dev-contest commented:
Hi @bbartling you have to to resolve the conflicts files and push the updated commit again. thanks

bbartling commented:
Hi @OpenVINO-dev-contest 

So I did make some fixes....but the conflicts still exist. Can do this on a Github project its where the conflicts exist:

1. README - I added a  number 8 and 9 (continuation from number 7) about how to run the IoT app
2. webcam.py - I added in the grid arg to my file and tested. But what is missing on your end is an additional arg to run the Flask web app and another arg if its okay I put in for confidence.
3.  yolov7.py - I added in the init the args for Flask web app and conf. In the `draw` method I also added some code to count people for when the detected objects are looped over the if its a person detected to count +1 which is then added to the web app for rest API `self.routes.net_people_count = people` and also a copy of the frame for the web app to render `self.routes.framecopy = img` lines 206/207. 

And I think that is about it from what I can see....should I do anything else on my end to resolve? Also still working on a JavaScript issue but need more time coming up with a solution...

OpenVINO-dev-contest commented:
Hi @bbartling I think you can fully git clone this repository in a new local fold, then manually add your update on it again. You should keep the a clean project firstly and then do your modification, otherwise some unexpected conflict may happen.

bbartling commented:
Hi @OpenVINO-dev-contest sorry newbie here on git....just so I am understanding correctly, should I delete my fork of your repo and start over with a fresh new fork to work off of and manually add things back together like you mentioned above?

OR

Should I keep my fork but clone your repository to my local desktop and basically start from scratch again off of my forked repo without reforking anything??

hopefully that makes sense, ha.

OpenVINO-dev-contest commented:
> Hi @OpenVINO-dev-contest sorry newbie here on git....just so I am understanding correctly, should I delete my fork of your repo and start over with a fresh new fork to work off of and manually add things back together like you mentioned above?
> 
> OR
> 
> Should I keep my fork but clone your repository to my local desktop and basically start from scratch again off of my forked repo without reforking anything??
> 
> hopefully that makes sense, ha.

I think the first opinion is a safer method.
